server/controllers/message.controller.js

```diff
@@ -37,18 +38,36 @@ function get(req, res) {
  * @returns {Message}
  */
 function send(req, res, next) {
-    const message = Message.build({
-        to: req.body.to,
-        from: req.body.from,
-        subject: req.body.subject,
-        message: req.body.message,
-        created: db.sequelize.fn('NOW'),
-    });
-    message.save()
-    .then(savedMessage => res.json(savedMessage))
-    .catch(e => next(e));
+    console.log(`length of recipient array-------${req.body.to.length}`);
+    // var array_lenth = req.body.to.length;
+    // console.log('array_length: '+array_length);
+
+    // defining a variable for array length wasn't working
+    let saveMessage = null;
+    for (let i = 0; i < req.body.to.length; i += 1) {
+        const message = Message.build({
+            to: req.body.to,
+            from: req.body.from,
+            subject: req.body.subject,
+            message: req.body.message,
+            owner: req.body.to[i],
+            created: db.sequelize.fn('NOW'),
+        });
+        //  message.save()
+        //  .then(savedMessage => res.json(savedMessage))
+        //  .catch(e => next(e));
+
+        saveMessage = message.save();
```

Comment:
Not sure what is going on here. How does this behave asynchronously with large batches of messages?